applicationset/generators/generator_spec_processor.go

```diff
@@ -78,6 +87,18 @@ func Transform(requestedGenerator argoprojiov1alpha1.ApplicationSetGenerator, al
 	return res, firstError
 }
 
+func keepOnlyStringValues(in map[string]interface{}) map[string]string {
```

Comment:
@crenshaw-dev  what do you think about it ?

Reply:
Ooh. Interesting problem. I think I'm okay with this limitation. We can always try to expand label matching support later.